Bio/SeqIO/__init__.py

```diff
@@ -726,16 +721,10 @@ def read(handle, format, alphabet=None):
     to read multiple records from the handle.
     """
     iterator = parse(handle, format, alphabet)
-    try:
-        first = next(iterator)
-    except StopIteration:
-        first = None
+    first = next(iterator, None)
     if first is None:
         raise ValueError("No records found in handle")
-    try:
-        second = next(iterator)
-    except StopIteration:
-        second = None
+    second = next(iterator, None)
     if second is not None:
         raise ValueError("More than one record found in handle")
```

Comment:
An alternative is
```
    try:
        record = next(iterator)
    except StopIteration:
        raise ValueError("No records found in handle") from None
    try:
        next(iterator)
        raise ValueError("More than one record found in handle")
    except StopIteration:
        pass
```
Then you don't need to define `second`.

Reply:
thanks- i did that- could do this actually?

```
    try:
        record = next(iterator)
    except StopIteration:
        raise ValueError("No records found in handle") from None
    try:
        next(iterator)
    except StopIteration:
        return record
    raise ValueError("More than one record found in handle")
```

Reply:
Yes, but I think I prefer Michiel's as the typical case drops though to the end of the function (rather than returning early).